btcec/ciphering.go

```diff
@@ -44,173 +38,5 @@ var (
 // public key using Diffie-Hellman key exchange (ECDH) (RFC 4753).
 // RFC5903 Section 9 states we should only return x.
 func GenerateSharedSecret(privkey *PrivateKey, pubkey *PublicKey) []byte {
-	x, _ := pubkey.Curve.ScalarMult(pubkey.X, pubkey.Y, privkey.D.Bytes())
-	return x.Bytes()
-}
-
-// Encrypt encrypts data for the target public key using AES-256-CBC. It also
-// generates a private key (the pubkey of which is also in the output). The only
-// supported curve is secp256k1. The `structure' that it encodes everything into
-// is:
-//
-//	struct {
-//		// Initialization Vector used for AES-256-CBC
-//		IV [16]byte
-//		// Public Key: curve(2) + len_of_pubkeyX(2) + pubkeyX +
-//		// len_of_pubkeyY(2) + pubkeyY (curve = 714)
-//		PublicKey [70]byte
-//		// Cipher text
-//		Data []byte
-//		// HMAC-SHA-256 Message Authentication Code
-//		HMAC [32]byte
-//	}
-//
-// The primary aim is to ensure byte compatibility with Pyelliptic.  Also, refer
-// to section 5.8.1 of ANSI X9.63 for rationale on this format.
-func Encrypt(pubkey *PublicKey, in []byte) ([]byte, error) {
```

Comment:
Most of the errors declared at the top of this file and no longer used and can be removed as well.